src/context/concrete_base.rs

```diff
@@ -185,6 +190,27 @@ impl ConcreteBaseAudioContext {
             LISTENER_PARAM_IDS.end,
         );
 
+        // (?) only for online context
+        if event_channel.is_some() {
+            // init event loop
+            let event_channel = event_channel.unwrap();
+            let event_handlers = base.inner.event_handlers.clone();
+            //
+            std::thread::spawn(move || loop {
```

Comment:
Instead of `loop` and `recv` you can use 
`for message in event_channel.iter() { `